crates/api_models/src/admin.rs

```diff
@@ -405,3 +405,31 @@ pub struct ToggleKVRequest {
     #[schema(example = true)]
     pub kv_enabled: bool,
 }
+
+#[derive(Debug, Clone, Default, Eq, PartialEq, serde::Deserialize, serde::Serialize, ToSchema)]
+pub struct MerchantConnectorDetailsWrap {
+    /// Creds Identifier is to uniquely identify the credentials. Do not send any sensitive info in this field.
+    pub creds_identifier: String,
+    /// Merchant connector details type type. Base64 Encode the credentials and send it in  this type and send as a string.
+    #[schema(value_type = Option<MerchantConnectorDetails>, example = r#"{
+        "connector_account_details": {
+            "auth_type": "HeaderKey",
+            "api_key":"sk_test_xxxxxexamplexxxxxx12345"
+        },
+        "metadata": {
+            "user_defined_field_1": "sample_1",
+            "user_defined_field_2": "sample_2", 
+        },
+    }"#)]
+    pub encoded_data: Option<String>,
```

Comment:
Will make `encoded_data`  masking::Secret<String>.
 
Since `creds_identifier` is anyhow going to be in the `return_url`, partial masking is not required right ?